### ISSUE TITLE
Option request: allow same folder imports (no-relative-import)

### ISSUE THREAD

**Issue by axelboc** [Microsoft/tslint-microsoft-contrib#493]
Angular generates specs with same folder imports, like:

```ts
import { MyComponent } from './my.component';
```

So it would be really convenient if rule `no-relative-import` had an option to ignore imports that point to the same folder as the current file, with `./`.

With this option enabled, the `no-relative-import` rule would still have to report unnecessarily complex same folder imports like `./../my/my.component`.

**Comment by IllusionMH** [comment]
May I take on this?

If I understand correctly - new option should be created.
Is there any preference for option name?
Candidates:
- `allowSameFolder`
- `allowSameFolderImport`
- `allowSiblings`

**Comment by JoshuaKGoldberg** [comment]
Absolutely @IllusionMH, thanks for taking this on!

I like `allowSiblings` the most personally.

**Comment by karol-majewski** [comment]
`allow-siblings` or `allow-same-folder` perhaps? This would match the kebab-case convention used elsewheere.

```json
"only-arrow-functions": [true, "allow-declarations", "allow-named-functions"]
```

**Comment by IllusionMH** [comment]
Sorry for pinging all in this thread, but in the process of implementation I had another question:

Should rule allow
```ts
import { MyComponent } from './subcomponents/sub.component';
```
when `allow-siblings` is enabled?

Or it might be another option to the rule?

P.S. Its really late here, so I'll hope to submit first WIP PR tomorrow.

**Comment by JoshuaKGoldberg** [comment]
That seems reasonable to allow by default, especially for projects that store assets inline with components:

```typescript
import { closeIcon } from "./my.component/assets";
```
